src/js/stores/MesosStore.js

```diff
@@ -366,7 +366,16 @@ AppDispatcher.register(function (action) {
       addDataToMap(data.id, storeData.taskFileMap, data.files.map(file => {
         var encodedPath = encodeURIComponent(file.path);
         file.host = data.host;
-        file.name = /[^/]+\/?$/.exec(file.path)[0];
+        /* The basename will be found in index 1 if it is a backslash
+           (Windows) path. The basename will be found in index 0 if it is
+           a forward slash path. A backslash path will also match in index 0,
+           but it will not contain the basename. */
+        var basename_match = /[a-zA-Z]:.*\\([^\\]+\\?$)|[^\/]+\/?$/.exec(
```

Comment:
The pattern and condition looks unnecessary complex. The following pattern should be sufficient to also extract the basename from Windows path. 

```js
/[^/\\]+[/\\]{0,2}$/
```

_Side Note: I wonder why decided to include ending slashes. It might makes sense to add a capture group to only get the name. `/([^/\\]+)[/\\]{0,2}$/.exec(file.path)[1]`_